src/System.IO.FileSystem/src/System/IO/FileStream.Unix.cs

```diff
@@ -16,5 +16,10 @@ public FileStream(SafeFileHandle handle, FileAccess access, int bufferSize, bool
         {
             this._innerStream = new UnixFileStream(handle, access, bufferSize, isAsync, this);
         }
+
+        private static void ValidatePath(string fullPath)
```

Comment:
You can avoid this by declaring private partial static void ValidatePath(string fullPath); in FileStream.cs and adding "partial" to the implementation in FileStream.Win32.cs.  That way no call to ValidatePath is added to the IL for unix build.